client/my-sites/sidebar/sidebar.jsx

```diff
@@ -681,10 +678,16 @@ export const MySitesSidebar = React.createClass( {
 	},
 
 	render: function() {
-		var publish = !! this.publish(),
-			appearance = ( !! this.themes() || !! this.menus() ),
-			configuration = ( !! this.sharing() || !! this.users() || !! this.siteSettings() || !! this.plugins() || !! this.upgrades() ),
-			vip = !! this.vip();
+		const showPublishMenu = !! this.renderPublishMenu();
+		const showAppearanceMenu = !! this.renderThemesOption() || !! this.renderMenusOption();
+		const showConfigurationMenu = (
+			!! this.renderSharingOption() ||
+			!! this.renderUsersOption() ||
+			!! this.renderSettingsOption() ||
+			!! this.renderPluginsOption() ||
+			!! this.renderUpgradesOption()
+		);
```

Comment:
Yep it's pretty nasty!
I didn't want to bring new checking methods in to the scope of this PR but I am planning to add some (see the checklist in #9054).

While theres some context can I run a thought by you? @gwwar too?

Should a jsx rendering method care much about privileges or conditions beyond guard clausing against errors?

In this file each `renderSomeComponent()` function does it's own checks.  
I usually prefer to do the checking separately first and just not even attempt to call this function (or component once it's moved out) if those checks aren't true.
We're sort of doing this here but in a weird inefficient way where we use a first pass as the check, then the second pass to generate the jsx.

The way I see it is that this function/component should be quite naive to wether or not it passes the logical conditions to render. Rather it should render when told to and only be told to when those conditions are met - the responsibility should come from the caller, not the component.

Any thoughts to this approach? - I'm conscious that I might be overlooking something! I'll open a PR in the near future that shows this a bit better than I can explain :P